From 696d957e13d1bc5b2034dcd637bc088cc4646a53 Mon Sep 17 00:00:00 2001
From: Adam Williamson <awilliam@redhat.com>
Date: Fri, 15 May 2020 09:15:20 -0700
Subject: [PATCH] Don't always expect base64.decodestring to exist

This does the same as before, but in such a way as not to fail if
`base64.decodestring` doesn't exist at all - as is the case in
Python 3.9.

This is a backport of #206 on top of 5.2.1.

Signed-off-by: Adam Williamson <awilliam@redhat.com>
---
 feedparser/feedparser.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/feedparser/feedparser.py b/feedparser/feedparser.py
index 999cb0d..1548db8 100644
--- a/feedparser/feedparser.py
+++ b/feedparser/feedparser.py
@@ -90,7 +90,10 @@ except ImportError:
     base64 = binascii = None
 else:
     # Python 3.1 deprecates decodestring in favor of decodebytes
-    _base64decode = getattr(base64, 'decodebytes', base64.decodestring)
+    try:
+        _base64decode = base64.decodebytes
+    except AttributeError:
+        _base64decode = base64.decodestring
 
 # _s2bytes: convert a UTF-8 str to bytes if the interpreter is Python 3
 # _l2bytes: convert a list of ints to bytes if the interpreter is Python 3
-- 
2.26.2

